contrib/IECoreUSD/src/IECoreUSD/PointInstancerAlgo.cpp

```diff
@@ -0,0 +1,139 @@
+	{
+		newPoints->variables["prototypeIndex"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Vertex, protoIndicesData );
+	}
+
+	if( auto idsData = DataAlgo::fromUSD( pointInstancer.GetIdsAttr(), time ) )
+	{
+		newPoints->variables["instanceId"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Vertex, idsData );
+	}
+
+	if( auto orientationData = DataAlgo::fromUSD( pointInstancer.GetOrientationsAttr(), time ) )
+	{
+		newPoints->variables["orientation"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Vertex, orientationData );
+	}
+
+	if( auto scaleData = DataAlgo::fromUSD( pointInstancer.GetScalesAttr(), time ) )
+	{
+		newPoints->variables["scale"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Vertex, scaleData );
+	}
+
+	if( auto velocityData = DataAlgo::fromUSD( pointInstancer.GetVelocitiesAttr(), time ) )
+	{
+		newPoints->variables["velocity"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Vertex, velocityData );
+	}
+
+#if USD_VERSION >= 1911
+	if( auto accelerationData = DataAlgo::fromUSD( pointInstancer.GetAccelerationsAttr(), time ) )
+	{
+		newPoints->variables["acceleration"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Vertex, accelerationData );
+	}
+#endif
+
+	if( auto angularVelocityData = DataAlgo::fromUSD( pointInstancer.GetAngularVelocitiesAttr(), time ) )
+	{
+		newPoints->variables["angularVelocity"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Vertex, angularVelocityData );
+	}
+
+	// Prototype paths
+
+	pxr::SdfPathVector targets;
+	pointInstancer.GetPrototypesRel().GetForwardedTargets( &targets );
+
+	IECore::StringVectorDataPtr prototypeRootsData = new IECore::StringVectorData();
+	auto &prototypeRoots = prototypeRootsData->writable();
+	prototypeRoots.reserve( targets.size() );
+	for( const auto &t : targets )
+	{
+		prototypeRoots.push_back( t.GetString() );
+	}
+
+	newPoints->variables["prototypeRoots"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Constant, prototypeRootsData );
+
+	return newPoints;
+}
+
+bool pointInstancerMightBeTimeVarying( pxr::UsdGeomPointInstancer &instancer )
+{
+	return
+		instancer.GetPositionsAttr().ValueMightBeTimeVarying() ||
+		instancer.GetProtoIndicesAttr().ValueMightBeTimeVarying() ||
+		instancer.GetIdsAttr().ValueMightBeTimeVarying() ||
+		instancer.GetOrientationsAttr().ValueMightBeTimeVarying() ||
+		instancer.GetScalesAttr().ValueMightBeTimeVarying() ||
+		instancer.GetVelocitiesAttr().ValueMightBeTimeVarying() ||
+#if USD_VERSION >= 1911
+		instancer.GetAccelerationsAttr().ValueMightBeTimeVarying() ||
+#endif
+		instancer.GetAngularVelocitiesAttr().ValueMightBeTimeVarying()
+	;
+}
+
+ObjectAlgo::ReaderDescription<pxr::UsdGeomPointInstancer> g_curvesReaderDescription( pxr::TfToken( "PointInstancer" ), readPointInstancer, pointInstancerMightBeTimeVarying );
```

Comment:
```suggestion
ObjectAlgo::ReaderDescription<pxr::UsdGeomPointInstancer> g_pointInstancerReaderDescription( pxr::TfToken( "PointInstancer" ), readPointInstancer, pointInstancerMightBeTimeVarying );
```

Reply:
Thanks - fixed in 62211cf. Turned out I'd done this in other spots too, so I've fixed those as well.